extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -126,5 +130,23 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding("hdfs").to(HdfsTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", HdfsTaskLogsConfig.class);
     binder.bind(HdfsTaskLogs.class).in(LazySingleton.class);
+
+    final String principal = props.getProperty("druid.hdfs.security.kerberos.principal");
+    final String keytab = props.getProperty("druid.hdfs.security.kerberos.keytab");
+    if (!Strings.isNullOrEmpty(principal) && !Strings.isNullOrEmpty(keytab)) {
+      UserGroupInformation.setConfiguration(conf);
+      if (UserGroupInformation.isSecurityEnabled()) {
+        try {
+          if (UserGroupInformation.getCurrentUser().hasKerberosCredentials() == false
+              || UserGroupInformation.getCurrentUser().getUserName() != principal) {
+            log.info("trying to authenticate user [%s] with keytab [%s]", principal, keytab);
+            UserGroupInformation.loginUserFromKeytab(principal, keytab);
```

Comment:
1. Generally this seems like the wrong place to do network operations. At this point we haven't even finished binding things for the injector yet, let alone used it to actually construct objects. It's more normal for druid stuff to defer runtime operations like that to start/stop rather than guice binding time or construction time.
2. Cool, sounds good.